test/box-luatest/gh_8051_set_box_cfg_thru_env_test.lua

```diff
@@ -0,0 +1,93 @@
+local server = require('luatest.server')
+local popen = require('popen')
+local clock = require('clock')
+
+local t = require('luatest')
+local g = t.group()
+
+g.after_all = function()
+    if g.server ~= nil then
+       g.server:stop()
+   end
+end
+
+local TARANTOOL_PATH = arg[-1]
+
+local function popen_run(command)
+    local cmd = TARANTOOL_PATH .. ' -i 2>&1'
+    local ph = popen.new({cmd}, {
+        shell = true,
+        setsid = true,
+        group_signal = true,
+        stdout = popen.opts.PIPE,
+        stderr = popen.opts.PIPE,
+        stdin = popen.opts.PIPE,
+    })
+    t.assert(ph, 'process is not up')
+
+    ph:write(command)
+
+    local output = '';
+    local time_quota = 2.0
+    local start_time = clock.monotonic();
+    while output:find('message') == nil
+        and clock.monotonic() - start_time < time_quota do
+        local data = ph:read({timeout = 1.0})
```

Comment:
I'm not a fan of tight timeouts -- it is often lead to flakiness in some environments (small machine, many parallel jobs). We can leave it as is, but I would set 20/10 seconds instead of 2/1.

Reply:
Set time_quota to 10.0, it seems to be the standard for other popen tests.